languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java

```diff
@@ -81,13 +81,13 @@ public String getDescription() {
         if (replacement == null) {
           replacement = getNumberReplacements().get(tokens[i].getToken());
           if (msg == null) {
-            msg = "Ba chóir duit <suggestion>" + tokens[i - 1].getToken() + " " + replacement + " " + tokens[i + 1].getToken() + "</suggestion> a scríobh";
+            msg = "Ba chóir duit <suggestion>" + replacement + "</suggestion> a scríobh";
           }
         }
       }
       if (msg != null) {
         RuleMatch match = new RuleMatch(
-          this, sentence, tokens[prevTokenIndex].getStartPos(), tokens[markEnd].getEndPos(), msg, "Uimhir phearsanta");
+          this, sentence, tokens[prevTokenIndex].getStartPos(), tokens[markEnd].getStartPos()+tokens[markEnd].getEndPos(), msg, "Uimhir phearsanta");
```

Comment:
`tokens[markEnd].getStartPos()+tokens[markEnd].getEndPos()` maybe I don't understand what you want here, but if thelast token is close to the sentence end position, adding something to `tokens[markEnd].getEndPos()` will also cause index out of bounds.
I am talking of this, after testing completed versions with the rule active. Without this, the rule can't be activated without crashing the program in the Irish language.

Reply:
getEndPos() gives the length of that token's match, not the overall, so it needs the start added from another token. I've put in a check to set it to 0 if it's in the same token

Reply:
You have looked at the internal better than I did, but, from experience, I believe that `getEndPos()` gives the numbered position in characters from the start of the sentence until the last character of the token. 
Anyway, please test the compiled version. I believe you will find it does what the tests expect (with a minor correction to the `beirt` suggestion).

Reply:
I was consistently getting assert errors that the getEndPos() result was less than getStartPos()